### ISSUE TITLE
[op-is-open] shortcode error

### ISSUE THREAD

**Issue by lorro** [janizde/WP-Opening-Hours#3]
Whether I do `echo do_shortcode('[op-is-open]');`

or type `[op-is-open]` in the visual editor I always get the following error:

`Catchable fatal error: Argument 1 passed to OpeningHours\Module\Shortcode\AbstractShortcode::renderShortcode() must be of the type array, string given in /home/srdev/public_html/wp-content/plugins/wp-opening-hours/classes/OpeningHours/Module/Shortcode/AbstractShortcode.php on line 119`

**Comment by waynep16** [comment]
How do we specify which branch we need to show is open ?

 op-is-open id=xx ???

**Comment by janizde** [comment]
It is the set's id (which also is the post id of the set post). I see, that it can be very confusing and I will add a Meta Box to the Set edit screen, showing which id to use

**Comment by waynep16** [comment]
Okay. So please can you tell me how to use the short code to show is open for a particular branch ?? 

It doesn't work for me and I get blank page error

**Comment by lorro** [comment]
Wayne,

these are the "default attributes" of the shortcode.
`$default_attributes = array(
			'set_id'              => null,
			'open_text'           => __( 'We\'re currently open.', self::TEXTDOMAIN ),
			'closed_text'         => __( 'We\'re currently closed.', self::TEXTDOMAIN ),
			'show_next'           => false,
			'next_format'         => __( 'We\'re open again on %2$s (%1$s) from %3$s to %4$s', self::TEXTDOMAIN ),
			'before_widget'       => null,
			'after_widget'        => null,
			'before_title'        => null,
			'after_title'         => null,
			'title'               => null,
			'classes'             => null,
			'next_period_classes' => null,
			'open_class'          => 'op-open',
			'closed_class'        => 'op-closed',
			'date_format'         => I18n::getDateFormat(),
			'time_format'         => I18n::getTimeFormat()
		);`

So you have to look up the id of the Branch you added and supply it with the shortcode
just give set_id = "xxx" as a param. [op-is-open set_id="111"] f.e.

**Comment by waynep16** [comment]
Thank you,works just as expected :)

**Comment by waynep16** [comment]
Hey Lorro, Can you please show me the parameters for  op-overview ?

**Comment by janizde** [comment]
Default Attributes for Overview Shortcode:
```
		$default_attributes = array(
			'before_title'             => '<h3 class="op-overview-title">',
			'after_title'              => '</h3>',
			'before_widget'            => '<div class="op-overview-shortcode">',
			'after_widget'             => '</div>',
			'set_id'                   => 0,
			'title'                    => null,
			'show_closed_days'         => false,
			'show_description'         => true,
			'highlight'                => 'nothing',
			'compress'                 => false,
			'short'                    => false,
			'include_io'               => false,
			'include_holidays'         => false,
			'caption_closed'           => __( 'Closed', I18n::TEXTDOMAIN ),
			'table_classes'            => null,
			'row_classes'              => null,
			'cell_classes'             => null,
			'cell_heading_classes'     => null,
			'cell_periods_classes'     => null,
			'cell_description_classes' => 'op-set-description',
			'highlighted_period_class' => 'highlighted',
			'highlighted_day_class'    => 'highlighted',
			'table_id_prefix'          => 'op-table-set-',
			'time_format'              => I18n::getTimeFormat(),
			'hide_io_date'             => false
		);

		$this->setDefaultAttributes( $default_attributes );

		$valid_attribute_values = array(
			'highlight'        => array( 'nothing', 'period', 'day' ),
			'show_closed_day'  => array( false, true ),
			'show_description' => array( true, false ),
			'include_io'       => array( false, true ),
			'include_holidays' => array( false, true ),
			'hide_io_date'     => array( false, true )
		);
```

**Comment by lorro** [comment]
Yes, I figured that out by now. I just forgot to supply the set_id parameter. Worked fine when I added that!

**Comment by janizde** [comment]
fixed in 5cf64ff999f9ec1ffa892fe17cbfdc138a5ec825
it will now just not render the shortcode if the set_id is not specified instead of throwing an error
